### PR TITLE
Install V2 Core and V2 Periphery

### LINKED ISSUE #9: Install V2 Core and V2 Periphery as submodules
Since the addition of the V2 Periphery examples (ref https://github.com/sablier-labs/examples/pull/4), the number of imports in this repo has skyrocketed.

Thus, it would be helpful to install V2 Core and V2 Periphery as git submodules, so that we can compile the contracts and get type errors.

### PR DESCRIPTION
**Pull request by andreivladbrg**
Closes https://github.com/sablier-labs/examples/issues/9

### CODE REVIEW

File: cache/solidity-files-cache.json
Comment (PaulRBerg) [inline cache/solidity-files-cache.json:1]: cache files should not be committed - run `git rm -r --cached ./cache` to delete this, and then push with force .

File: src/v2/periphery/BatchLockupDynamicStreamCreator.sol [hunk 1]
```diff
@@ -71,20 +71,20 @@ contract BatchLockupDynamicStreamCreator {
         Batch.CreateWithMilestones memory stream0;
         stream0.sender = address(proxy); // The sender will be able to cancel the stream
         stream0.recipient = address(0xcafe); // The recipient of the streamed assets
-        stream0.totalAmount = uint128(totalAmount); // The total amount of each stream, inclusive of all fees
+        stream0.totalAmount = uint128(transferAmount); // The total amount of each stream, inclusive of all fees
```
Comment (PaulRBerg) [inline src/v2/periphery/BatchLockupDynamicStreamCreator.sol]: Should be `perStreamAmount`.

File: src/v2/periphery/BatchLockupDynamicStreamCreator.sol [hunk 2]
```diff
@@ -94,20 +94,20 @@ contract BatchLockupDynamicStreamCreator {
         Batch.CreateWithMilestones memory stream1;
         stream1.sender = address(proxy); // The sender will be able to cancel the stream
         stream1.recipient = address(0xbeef); // The recipient of the streamed assets
-        stream1.totalAmount = uint128(totalAmount); // The total amount of each stream, inclusive of all fees
+        stream1.totalAmount = uint128(transferAmount); // The total amount of each stream, inclusive of all fees
```
Comment (PaulRBerg) [inline src/v2/periphery/BatchLockupDynamicStreamCreator.sol]: Should be `perStreamAmount`

File: src/v2/periphery/BatchLockupLinearStreamCreator.sol [hunk 3]
```diff
@@ -71,7 +71,7 @@ contract BatchLockupLinearStreamCreator {
         Batch.CreateWithDurations memory stream0;
         stream0.sender = address(proxy); // The sender will be able to cancel the stream
         stream0.recipient = address(0xcafe); // The recipient of the streamed assets
-        stream0.totalAmount = uint128(totalAmount); // The total amount of each stream, inclusive of all fees
+        stream0.totalAmount = uint128(transferAmount); // The total amount of each stream, inclusive of all fees
```
Comment (PaulRBerg) [inline src/v2/periphery/BatchLockupLinearStreamCreator.sol]: Should be `perStreamAmount`.

File: src/v2/periphery/BatchLockupLinearStreamCreator.sol [hunk 4]
```diff
@@ -83,7 +83,7 @@ contract BatchLockupLinearStreamCreator {
         Batch.CreateWithDurations memory stream1;
         stream1.sender = address(proxy); // The sender will be able to cancel the stream
         stream1.recipient = address(0xbeef); // The recipient of the streamed assets
-        stream1.totalAmount = uint128(totalAmount); // The total amount of each stream, inclusive of all fees
+        stream1.totalAmount = uint128(transferAmount); // The total amount of each stream, inclusive of all fees
```
Comment (PaulRBerg) [inline src/v2/periphery/BatchLockupLinearStreamCreator.sol]: Should be `perStreamAmount`.

**Review by PaulRBerg — CHANGES_REQUESTED** [review]
There's no need to add a new `src` directory - we can just point `src` to `v2` in `foundry.toml`. I'll push a commit to do this.

File: cache/solidity-files-cache.json
Comment (andreivladbrg) [reply cache/solidity-files-cache.json:1]: Yeah, this is why I added the `.gitignore` file
https://github.com/sablier-labs/examples/blob/a261314f05e5cd6b0b67642eacf6ec1f3dfa1b32/.gitignore#L2

Might have pushed it by mistake

**Review by PaulRBerg — APPROVED** [review]
LGTM now

**Comment by PaulRBerg** [comment]
Thanks for merging @andreivladbrg - in the future, can we squash & merge when there are commits of the sort "fix: address PR feedback"?

**Comment by andreivladbrg** [comment]
>  can we squash & merge when there are commits

I actually used squash & merge, see [here](https://github.com/sablier-labs/examples/commit/eae781627e15d963a62c1545fe0944ad94d5519c).

>   "fix: address PR feedback"

Do you mean to actually remove the message?

**Comment by PaulRBerg** [comment]
> I actually used squash & merge, see [here](https://github.com/sablier-labs/examples/commit/eae781627e15d963a62c1545fe0944ad94d5519c).

Oh, sorry! Good job

> Do you mean to actually remove the message?

No

**Pull request merged**